cellprofiler/modules/plugins/imagetemplate.py

```diff
@@ -176,128 +276,110 @@ def create_settings(self):
+
+        super(ImageTemplate, self).run(workspace)
 
     #
-    # display lets you use matplotlib to display your results.
+    # "volumetric" indicates whether or not this module supports 3D images.
+    # The "gradient_image" function is inherently 2D, and we've noted this
+    # in the documentation for the module. Explicitly return False here
+    # to indicate that 3D images are not supported.
     #
-    def display(self, workspace, figure):
-        #
-        # the "figure" is really the frame around the figure. You almost always
-        # use figure.subplot or figure.subplot_imshow to get axes to draw on
-        # so we pretty much ignore the figure.
-        #
-        figure = workspace.create_or_find_figure(subplots=(3, 1))
-        #
-        # Show the user the input image
-        #
-        figure.subplot_imshow_grayscale(
-                0, 0,  # show the image in the first row and column
-                workspace.display_data.input_pixels,
-                title=self.input_image_name.value)
-        lead_subplot = figure.subplot(0, 0)
-        #
-        # Show the user the gradient image, linking it to the first
-        # so that they zoom and pan together
-        #
-        figure.subplot_imshow_grayscale(
-                1, 0,  # show the image in the first row and second column
-                workspace.display_data.gradient,
-                title="Gradient",
-                sharex=lead_subplot, sharey=lead_subplot)
-        #
-        # Show the user the final image
-        #
-        figure.subplot_imshow_grayscale(
-                2, 0,  # show the image in the first row and last column
-                workspace.display_data.output_pixels,
-                title=self.output_image_name.value,
-                sharex=lead_subplot, sharey=lead_subplot)
+    def volumetric(self):
+        return False
+
+#
+# This is the function that gets called during "run" to create the output image.
+# The first parameter must be the input image data. The remaining parameters are
+# the additional settings defined in "settings", in the order they are returned.
+#
+# This function must return the output image data (as a numpy array).
+#
+def gradient_image(pixels, gradient_choice, automatic_smoothing, scale):
+    #
+    # Get the smoothing parameter
+    #
+    if automatic_smoothing:
+        # Pick the mode of the power spectrum - obviously this
+        # is pretty hokey, not intended to really find a good number.
+        #
+        fft = numpy.fft.fft2(pixels)
+        power2 = numpy.sqrt((fft * fft.conjugate()).real)
+        mode = numpy.argwhere(power2 == power2.max())[0]
+        scale = numpy.sqrt(numpy.sum((mode + .5) ** 2))
+
+    gradient_magnitude = scipy.ndimage.gaussian_gradient_magnitude(pixels, scale)
+
+    if gradient_choice == GRADIENT_MAGNITUDE:
+        gradient_image = gradient_magnitude
+    else:
+        # Numpy uses i and j instead of x and y. The x axis is 1
```

Comment:
I'm confused by this comment, where do `i` and `j` come up in this example?

Reply:
Oof. I'll clarify.